### PR TITLE
Migrate to Jotai v2 API

### PR DESCRIPTION
**Pull request by dai-shi**
Ref: https://github.com/pmndrs/jotai/pull/1515

No ETA to merge this yet.

### CODE REVIEW

**Comment by barelyhuman** [comment]
Want this to go as a minor bump or major bump release?

File: package.json [hunk 1]
```diff
@@ -90,10 +91,11 @@
     "webpack": "^5.72.0",
     "webpack-cli": "^4.9.2",
     "webpack-dev-server": "^4.8.1",
+    "yarn": "^1.22.19",
```
Comment (barelyhuman) [inline package.json]: Was this dep added, intentionally?

File: package.json [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "@types/react-dom": "^18.0.1",
     "@typescript-eslint/eslint-plugin": "^5.20.0",
     "@typescript-eslint/parser": "^5.20.0",
+    "add": "^2.0.6",
```
Comment (barelyhuman) [inline package.json]: Was this dep added, intentionally?

File: package.json (hunk 1 above)
Comment (dai-shi) [reply package.json]: Oh, no. It's caused by a typo. Let me fix it.

**Comment by dai-shi** [comment]
> Want this to go as a minor bump or major bump release?

It's up to you. For other jotai-* packages, I would probably do a minor bump if they are still on v0.*.

**Pull request merged**